### PR TITLE
[pythonic resources] [dagster-slack] Migrate resources

### PR DESCRIPTION
**Pull request by yuhan**

### CODE REVIEW

**Comment by yuhan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13464** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13464" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13464?utm_source=stack-comment).

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py [hunk 1]
```diff
@@ -1,24 +1,59 @@
-from dagster import Field, StringSource, resource
+from dagster import resource
+from dagster._config.structured_config import ConfigurableResource
+from pydantic import Field as PyField
```
Comment (schrockn) [inline python_modules/libraries/dagster-slack/dagster_slack/resources.py:3]: don't do PyField aliasing

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py [hunk 2]
```diff
@@ -1,24 +1,59 @@
...
 from slack_sdk.web.client import WebClient
 
 
+class SlackResource(ConfigurableResource):
+    """This resource is for connecting to Slack.
+
+    The resource object is a `slack_sdk.WebClient`.
```
Comment (schrockn) [inline python_modules/libraries/dagster-slack/dagster_slack/resources.py:10]: I don't think we should use the term "resource object" as that isn't a term of art in the system now.

**Review by schrockn — CHANGES_REQUESTED** [review]

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py [hunk 3]
```diff
@@ -1,29 +1,63 @@
-from dagster import Field, StringSource, resource
+from dagster import ConfigurableResource, resource
+from pydantic import Field
 from slack_sdk.web.client import WebClient
 
 
+class SlackResource(ConfigurableResource):
+    """This resource is for connecting to Slack.
+
+    The resource interacts with Slack via `slack_sdk.WebClient`.
```
Comment (schrockn) [inline python_modules/libraries/dagster-slack/dagster_slack/resources.py:9]: Also I think this language is slightly off. I returns a `slack_sdk.WebClient` from `get_client`. This sentence makes it sound like it calls the web client's methods on your behalf, which it doesn't.

**Review by schrockn — APPROVED** [review]

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py (hunk 3 above)
Comment (yuhan) [reply python_modules/libraries/dagster-slack/dagster_slack/resources.py:9]: im going to get rid of this line and move it to down to docstring of `get_client`

**Pull request merged**